remotes/docker/resolver.go

```diff
@@ -533,6 +559,9 @@ func (r *request) do(ctx context.Context) (*http.Response, error) {
 	}
 
 	ctx = log.WithLogger(ctx, log.G(ctx).WithField("url", u))
+	if r.trace {
+		ctx = httptrace.WithClientTrace(req.Context(), tracer)
```

Comment:
do we lose the `ctx` set in 561? will the `log.G` in lines 565 and 573 still work? (since we lost the `ctx` from line 561)

(curious!)

Reply:
yes, we do 😇 I can't remember what I was playing around with while testing, but at some point I changed this from `ctx` to `req.Context()` and never changed it back. Good catch!

Reply:
woohoo!!! i actually found something :) yay me!